LUA/ak/io/AkStatistik.lua

```diff
@@ -352,58 +483,69 @@ local function fillApiV1(dataKeys)
     data["api-entries"] = dataEntries
 end
 
+--- Initialize data.
+-- do it once
 local function initialize()
-	print("AkStatistik: init")
+    print("AkStatistik: init")
 
-	-- prepare data once
-	registerSignals()
-	registerSwitches()
+    -- prepare data once
+    registerSignals()
+    registerSwitches()
     registerTracks()
-	registerStructures()
+    registerStructures()
 
     -- export data once
-	fillEEPVersion()
+    fillEEPVersion()
 end
 
 local i = -1
 local writeLater = {}
+--- Main function of this module.
+-- Call this function in main loop of EEP. 
+-- do it frequently
+-- @param modulus Repetion frequency (1: every 200 ms, 5: every second, ...)
 function AkStatistik.statistikAusgabe(modulus)
--- call this function in main loop 
-
- 	-- default value for optional parameter
+   -- default value for optional parameter
    if not modulus or type(modulus) ~= "number" then
         modulus = 5
     end
 
-	-- initialization in first call
+    -- initialization in first call
     if i == -1 then
         initialize();
     end
     i = i + 1
 
-	-- process commands 
+    -- process commands 
     AkWebServerIo.processNewCommands()
 
     -- export data regularly
     if i % modulus == 0 then
-        local t0 = os.clock()	-- milliseconds precision
-		
-		fillTime()
-		fillStructures()
+        local t0 = os.clock() -- milliseconds precision
+        
+        -- Skip if the Web Server is not ready
+        -- Should be accept "ready" only or should be accept "not ready" too? 
+        -- (The Web Server might have finished reading the previous version of the file until filling the data structures again.)
+        if (AkStatistik.checkServerStatus and AkWebServerIo.checkWebServer() == 1) then    -- -1: not listening, 0: not ready, 1: ready
```

Comment:
Es hat mich einige Zeit gekostet 😃, aber hier ist ein Vorzeichenfehler drin. So wie der Code hier steht, wird die JSON-Datei genau dann geschrieben, wenn der Server nicht lauscht oder noch nicht fertig ist.

Entweder muss bei "ungleich 1" das return stattfinden oder den Code der hinter dem if steht sollte statt dem return im if ausgeführt werden (das würde ich bevorzugen, da returns vor dem Methoden-Ende es schwierig machen, den Code zu verstehen).

Wie wäre statt "AkWebServerIo.checkWebServer() == 1" eine Methode, z.B. "AkWebServerIo.shallUpdateJsonFile()", die true zurückgibt, wenn das JSON-File geschrieben werden soll? - Das erleichtert die Lesbarkeit und man muss beim Aufruf nicht wissen, dass es -1, 0 und 1 als Werte gibt.

Reply:
Weiter unten in der Methode gibt es die Zeile noch mal.

Reply:
Am besten wäre es doch, die Bedingung gleich in das Modulus-if reinzuschreiben, dann braucht man das return und dann stimmt auch der Vergleich mit "== 1":
```lua
if i % modulus == 0 and AkStatistik.checkServerStatus and AkWebServerIo.checkWebServer() == 1 then
   -- Ausgabe hier ...
end
```

Reply:
Test der neuen Version mit/ohne aktiven EEP-Web-Server ok
```
--clearlog()		-- call before loading AKStatistik does not clear the log file
local AkStatistik = require("ak.io.AkStatistik")
clearlog()			-- call after loading AKStatistik clears the log file, too
print('AkStatistik Version ' .. AkStatistik.programVersion)    

-- Allow to update the json file without checking if the Web Server is ready
AkStatistik.checkServerStatus = false -- true/false

function EEPMain()
	AkStatistik.statistikAusgabe(1)
	return 1
end
```